### PR TITLE
🎉 Source TikTok Marketing: support oauth flow

### LINKED ISSUE #7098: Source TikTok Marketing: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application.

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this:

If the connector already supports some auth mechanism like api_key, I suggest that this be a oneof nested inside a top-level field called "authentication":

{ authentication: { type: object oneOf: [ // api key, // oauth ] } }

If the connector only supports webflow oauth, then no changes are needed to the properties format and we will only need to add annotations.

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change.

## Acceptance criteria
The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
Oauth properties are annotated properly. See this PR for an example

### PR DESCRIPTION
**Pull request by annalvova05**
## What
Add TikTok Marketing OAuth support.
tiktok_1
tiktok_2

## How
Change spec.py and source.py. Add java part.

## Recommended reading order
1. spec.py
2. source.py
3. TikTokMarketingOAuthFlow.java

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/7636)

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/source.py [hunk 1]
```diff
@@ -50,17 +52,35 @@ def spec(self, *args, **kwargs) -> ConnectorSpecification:
...
     @staticmethod
     def _prepare_stream_args(config: Mapping[str, Any]) -> Mapping[str, Any]:
         """Converts an input configure to stream arguments"""
+        credentials = config.get("credentials") or config
+
+        access_token = credentials["access_token"]
+        secret = credentials.get("secret") or credentials.get("environment", {}).get("secret")
+        app_id = int(credentials.get("app_id", 0) or credentials.get("environment", {}).get("app_id", 0))
+        advertiser_id = int(credentials.get("advertiser_id", 0) or credentials.get("environment", {}).get("advertiser_id", 0))
+
```
Comment (antixar) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/source.py:77]: Could you split this parameters' parsing to 2 diff parts: parsing old and new config structure separately ?

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlow.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  protected String formatConsentUrl(final UUID definitionId,
+                                    final String appId,
+                                    final String redirectUrl,
+                                    final JsonNode inputOAuthConfiguration)
+      throws IOException {
+
+    final URIBuilder builder = new URIBuilder()
+        .setScheme("https")
+        .setHost("ads.tiktok.com")
+        .setPath("marketing_api/auth")
```
Comment (antixar) [inline airbyte-oauth/src/main/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlow.java:56]: why does this logic use "ads.tiktok.com" here but for loading of access tokens another one (business-api.tiktok.com)?
I remember that ads.tiktok.com is the old API domain. Please check

**Review by antixar — COMMENTED** [review]
Some minor comments

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlow.java (hunk 2 above)
Comment (annalvova05) [reply airbyte-oauth/src/main/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlow.java:56]: As I can see in the [documentation](https://ads.tiktok.com/marketing_api/docs?rid=esb88xalm6m&id=1709207085043713), TikTok changed its hostname only for API (`ads.tiktok.com/open_api` -> `business-api.tiktok.com/open_api`). Also, there is an Advertiser authorization URL in TikTok [Apps](https://ads.tiktok.com/marketing_api/apps) Basic information, it starts from `ads.tiktok.com`.

**Review by antixar — APPROVED** [review]
LGTM
Please publish your changes

**Pull request merged**